README.md

```diff
@@ -218,9 +296,79 @@ res.render("login", {
       size,
       color,
     }),
-  card: Card({
-    title: "Login With Ory",
-    children: Message({ message: "Woah there", severity: "info" }),
+  card: UserAuthCard({
+    title: !(flow.refresh || flow.requested_aal === "aal2")
+      ? "Sign In"
+      : "Two-Factor Authentication",
+    flow: flow as SelfServiceFlow,
+    flowType: "login",
+    cardImage: "ory-logo.svg",
+    additionalProps: {
+      forgotPasswordURL: "recovery",
+      signupURL: initRegistrationUrl,
+      logoutURL: logoutUrl,
+    },
   }),
 })
 ```
+
+---
+
+### Component System
+
+Elements solely rely on React components since they are easy to write and
+provides support to a large React based ecosystem. The project then bundles
+these components to their respective needs. An example is bundling for Preact
+which you can find in the
+[packages](https://github.com/ory/elements/tree/main/packages) folder. It uses
+the react components directly in this case but bundles it specifically for
+Preact support.
+
+Each component relies on a some CSS, which are located in the
+[theme](https://github.com/ory/elements/tree/main/src/theme) directory. To
+understand how this works, please refer to the [CSS System](#css-system)
+
+#### ExpressJs systems
+
+Express is an edge-case which requires the component to be wrapped by
+`ReactDOMServer` to produce only HTML. Each component needs to be wrapped by
+`ComponentWrapper` which essentially uses `ReactDOMServer`. The
+`elements-markup` package then bundles the React library with it so that the
+React code lives with the component library.
+
+Here is an example of exporting the `UserAuthCard`.
+
+```ts
+import {
+  UserAuthCard as userAuthCard,
+  UserAuthCardProps,
+} from "../react-components"
+
+export const UserAuthCard = (props: UserAuthCardProps) => {
+  return ComponentWrapper(userAuthCard(props))
+}
+
+export type { UserAuthCardProps } from "../react-components"
+```
+
+### Asset System
+
+Assets are bundled into a singular `style.css` file under each pacakges `dist/`
+folder. Anything placed inside the
+[assets](https://github.com/ory/elements/tree/main/src/assets) folder will be
```

Comment:
Do we have tree shaking enabled in these parts?

Not really related to this PR, just out of curiosity. It sounds like we are just copying all the files in this folder to one CSS files.

Reply:
they get bundled through Vite with minification. But yeah we need to do some sort of tree-shaking or chunking later. Since you most likely don't want all fonts and all styles at the same time.